plugins/modules/gitlab_merge_request.py

```diff
@@ -0,0 +1,415 @@
+        state_filter=dict(type='str', default="opened", choices=["opened", "closed", "locked", "merged"]),
+        assignee_ids=dict(type='str', required=False),
+        reviewer_ids=dict(type='str', required=False),
+        state=dict(type='str', default="present", choices=["absent", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['api_username', 'api_token'],
+            ['api_username', 'api_oauth_token'],
+            ['api_username', 'api_job_token'],
+            ['api_token', 'api_oauth_token'],
+            ['api_token', 'api_job_token'],
+            ['description', 'description_path'],
+        ],
+        required_together=[
+            ['api_username', 'api_password'],
+        ],
+        required_one_of=[
+            ['api_username', 'api_token', 'api_oauth_token', 'api_job_token']
+        ],
+        required_if=[
+            ['state', 'present', ['source_branch', 'target_branch', 'title'], True],
+            ['state', 'absent', ['source_branch', 'target_branch', 'title'], True],
+        ],
+        supports_check_mode=True
+    )
+    ensure_gitlab_package(module)
+
+    project = module.params['project']
+    source_branch = module.params['source_branch']
+    target_branch = module.params['target_branch']
+    title = module.params['title']
+    description = module.params['description']
+    labels = module.params['labels']
+    description_path = module.params['description_path']
+    remove_source_branch = module.params['remove_source_branch']
+    state_filter = module.params['state_filter']
+    assignee_ids = module.params['assignee_ids']
+    reviewer_ids = module.params['reviewer_ids']
+    state = module.params['state']
+
+    gitlab_version = gitlab.__version__
+    if LooseVersion(gitlab_version) < LooseVersion('2.3.0'):
+        module.fail_json(msg="community.general.gitlab_merge_request requires python-gitlab Python module >= 2.3.0 (installed version: [%s])."
+                             " Please upgrade python-gitlab to version 2.3.0 or above." % gitlab_version)
+
+    gitlab_instance = gitlab_authentication(module)
+    this_gitlab = GitlabMergeRequest(module=module, project=project, gitlab_instance=gitlab_instance)
+
+    r_source_branch = this_gitlab.get_branch(source_branch)
+    if not r_source_branch:
+        module.fail_json(msg="Source branch {b} not exist.".format(b=r_source_branch))
+
+    r_target_branch = this_gitlab.get_branch(target_branch)
+    if not r_target_branch:
+        module.fail_json(msg="Destination branch {b} not exist.".format(b=r_target_branch))
+
+    this_mr = this_gitlab.get_mr(title, source_branch, target_branch, state_filter)
+
+    if state == "present":
+        if description_path:
+            try:
+                f = open(description_path, 'rb')
+                description = to_text(f.read(), errors='surrogate_or_strict')
+                f.close()
```

Comment:
A `with` syntax would be safer if IOError occurs
```suggestion
                with open(description_path, 'rb') as f:
                    description = to_text(f.read(), errors='surrogate_or_strict')
```